starlette/schemas.py

```diff
@@ -65,14 +77,35 @@ def get_endpoints(
                     endpoints_info.append(
                         EndpointInfo(route.path, method.lower(), func)
                     )
+
         return endpoints_info
 
-    def parse_docstring(self, func_or_method: typing.Callable) -> dict:
+    def _extract_yaml(self, docstring: str) -> str:
+        """
+        We support having regular docstrings before the schema definition.
+
+        Here we return just the schema part from a docstring.
+        """
+        return docstring.split("---")[-1]
+
+    def parse_docstring(self, func_or_method: typing.Callable) -> typing.Optional[dict]:
```

Comment:
Would it make sense to return an empty dictionary in cases where we couldn't extract any info?

Reply:
Yes, that looks cleaner. Another question is whether we want to exclude those cases from the final result (`if not parsed:` below). I think it makes sense.

Reply:
Actually that makes sense yup. Perhaps we'll also be able to move towards dropping the `include_in_schema` argument in routing?

Reply:
Yes, maybe it's not longer necessary. I haven't used it myself, but maybe there are still cases when you want to explicitly disable some endpoints. But then you wouldn't use yaml syntax ...

Reply:
I guess in that case you could have something that looks like yaml but it isn't yaml, for example:

```python
@app.route("/foo")
def foo(request):
    """
    This endpoint does the following:
        - Receive input
        - Some magic
        - Return an output
    """
    pass  # pragma: no cover
```